data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/prepper/peerforwarder/PeerForwarderUtils.java

```diff
@@ -6,45 +6,13 @@
 package com.amazon.dataprepper.plugins.prepper.peerforwarder;
 
 import com.amazon.dataprepper.model.trace.Span;
-import com.linecorp.armeria.internal.shaded.bouncycastle.util.encoders.Hex;
-import io.opentelemetry.proto.trace.v1.InstrumentationLibrarySpans;
-import io.opentelemetry.proto.trace.v1.ResourceSpans;
 
-import java.util.AbstractMap;
-import java.util.ArrayList;
 import java.util.Collection;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 public final class PeerForwarderUtils {
-    public static int getResourceSpansSize(final ResourceSpans rs) {
-        return rs.getInstrumentationLibrarySpansList().stream().mapToInt(InstrumentationLibrarySpans::getSpansCount).sum();
-    }
-
-    public static List<Map.Entry<String, ResourceSpans>> splitByTrace(final ResourceSpans rs) {
-        final List<Map.Entry<String, ResourceSpans>> result = new ArrayList<>();
-        for (final InstrumentationLibrarySpans ils: rs.getInstrumentationLibrarySpansList()) {
-            final Map<String, ResourceSpans.Builder> batches = new HashMap<>();
-            for (final io.opentelemetry.proto.trace.v1.Span span: ils.getSpansList()) {
-                final String sTraceId = Hex.toHexString(span.getTraceId().toByteArray());
-                if (!batches.containsKey(sTraceId)) {
-                    final ResourceSpans.Builder newRSBuilder = ResourceSpans.newBuilder()
-                            .setResource(rs.getResource());
-                    newRSBuilder.addInstrumentationLibrarySpansBuilder().setInstrumentationLibrary(ils.getInstrumentationLibrary());
-                    batches.put(sTraceId, newRSBuilder);
-                }
-
-                // there is only one instrumentation library per batch
-                batches.get(sTraceId).getInstrumentationLibrarySpansBuilder(0).addSpans(span);
-            }
-
-            batches.forEach((traceId, rsBuilder) -> result.add(new AbstractMap.SimpleEntry<>(traceId, rsBuilder.build())));
-        }
-
-        return result;
-    }
 
     public static Map<String, List<Span>> splitByTrace(final Collection<Span> spans) {
```

Comment:
I believe this is no longer needed and this entire class can be eliminated.

Reply:
This one is still used in PeerForwarder